tests/drivers/build_all/sensors_a_m.conf

```diff
@@ -18,11 +18,11 @@ CONFIG_DHT=y
 # CONFIG_FXOS8700=y
 CONFIG_HDC1008=y
 CONFIG_HMC5883L=y
-CONFIG_HTS221=y
+# CONFIG_HTS221=y
 CONFIG_HP206C=y
```

Comment:
Can you check this is not the case anymore on recent master branch?
I've reverted some Kconfig change for these sensor to remove the dependency on disco board.

Reply:
Do you mean 2867b86 and 11eac23? Yes, these are included. The issue is that when I select `HAS_DTS_I2C_DEVICE` on a board that does not have hts221, the compile fails because `CONFIG_HTS221_NAME` and `CONFIG_HTS221_I2C_MASTER_DEV_NAME` are not defined. Same is true for lis3mdl.

Reply:
I see. But then once we end up with all sensors converted to DTS, we'll have the issue with all the sensors. If possible, I think this dependency should be fixed rather than by-passed. I'll have a look

Reply:
OK, I see now.
So this will be the case for all sensors converted to DTS not present on a board enabling HAS_DTS_I2C_DEVICE.
This is because HAS_DTS_I2C_DEVICE will prevent from inheriting of sensor's Kconfig default values.
So this is a general incompatibility of dts and "build all on all boards" kind of tests. Since dts will be more and more present, this is worse being addressed or at least discussed in a dedicated issue.